mlflow/pyfunc/__init__.py

```diff
@@ -941,7 +1029,91 @@ def predict(*args):
         else:
             return result[result.columns[0]]
 
-    udf = pandas_udf(predict, result_type)
+    result_type_hint = pd.DataFrame if isinstance(result_type, SparkStructType) else pd.Series
+
+    @pandas_udf(result_type)
+    def udf(
+        iterator: Iterator[Tuple[Union[pd.Series, pd.DataFrame], ...]]
+    ) -> Iterator[result_type_hint]:
```

Comment:
CC @HyukjinKwon

The `pandas_udf` "functionType" argument will be deprecated and removed soon.
So defining pandas iterator style UDF we need define type hint like this.

Reply:
Thank you!